crates/bevy_ecs/src/core/access.rs

```diff
@@ -219,63 +236,153 @@ impl<T: Hash + Eq + PartialEq + Copy> TypeAccess<T> {
         for write in writes {
             type_access.add_write(write);
         }
-
         for read in reads {
             type_access.add_read(read);
         }
-
         type_access
     }
 
     pub fn is_compatible(&self, other: &TypeAccess<T>) -> bool {
-        self.writes.is_disjoint(&other.reads_and_writes)
-            && self.reads_and_writes.is_disjoint(&other.writes)
+        if self.reads_all {
+            other.writes.is_empty()
+        } else if other.reads_all {
+            self.writes.is_empty()
+        } else {
+            self.writes.is_disjoint(&other.reads_and_writes)
+                && self.reads_and_writes.is_disjoint(&other.writes)
+        }
     }
 
     pub fn get_conflict<'a>(&'a self, other: &'a TypeAccess<T>) -> Option<&'a T> {
-        let conflict = self.writes.intersection(&other.reads_and_writes).next();
-        if conflict.is_some() {
-            return conflict;
+        if self.reads_all {
+            other.writes.iter().next()
+        } else if other.reads_all {
+            self.writes.iter().next()
+        } else {
+            match self.writes.intersection(&other.reads_and_writes).next() {
+                Some(element) => Some(element),
+                None => other.writes.intersection(&self.reads_and_writes).next(),
+            }
         }
-        self.reads_and_writes.intersection(&other.writes).next()
     }
 
-    pub fn union(&mut self, other: &TypeAccess<T>) {
+    pub fn extend(&mut self, other: &TypeAccess<T>) {
+        self.reads_all = self.reads_all || other.reads_all;
         self.writes.extend(&other.writes);
-        self.reads.extend(&other.reads);
         self.reads_and_writes.extend(&other.reads_and_writes);
     }
 
     pub fn add_read(&mut self, ty: T) {
         self.reads_and_writes.insert(ty);
-        self.reads.insert(ty);
     }
 
     pub fn add_write(&mut self, ty: T) {
         self.reads_and_writes.insert(ty);
         self.writes.insert(ty);
     }
 
+    pub fn read_all(&mut self) {
```

Comment:
It doesn't look like this is used anywhere. Is there a good reason to keep it around?

Reply:
It's intended to be used with `&World` and `&Resources` system parameters - as in, for systems that want full read access to those collections. Since they don't want exclusive access they can, in theory, be parallelized. Implementation is omitted to combat PR scope creep; a follow-up PR with that should be a rather low-hanging fruit.

Reply:
Makes sense to me. That seems useful and its worth keeping it around.